internal/statemachine/helper.go

```diff
@@ -619,29 +621,34 @@ func dpkgDivert(baseDir string, target string) (*exec.Cmd, *exec.Cmd) {
 	return execCommand("chroot", divert...), execCommand("chroot", undivert...)
 }
 
-// backupReplaceStartStopDaemon backup start-stop-daemon and replace it with a fake one
-// Returns a restore function to put the original one in place
-func backupReplaceStartStopDaemon(baseDir string) (func(error) error, error) {
-	const startStopDaemonContent = `#!/bin/sh
-echo
-echo "Warning: Fake start-stop-daemon called, doing nothing"
-`
+// divertExecWithFake replaces a target file in targetDir with a provided fake content
+// using dpkg-divert, and returns two commands lists: one for diverting, one for undiverting.
+func divertExecWithFake(targetDir string, file string, fakeContent string) ([]*exec.Cmd, []*exec.Cmd) {
```

Comment:
nitpick: The name is a bit misleading, especially since we have other `divertXXX` functions only diverting (and not also replacing the content).